crates/proof-of-sql/src/sql/postprocessing/group_by_postprocessing.rs

```diff
@@ -169,6 +177,153 @@ impl GroupByPostprocessing {
     }
 }
 
+impl<S: Scalar> PostprocessingStep<S> for GroupByPostprocessing {
+    /// Apply the group by transformation to the given `OwnedTable`.
+    fn apply(&self, owned_table: OwnedTable<S>) -> PostprocessingResult<OwnedTable<S>> {
+        // First evaluate all the aggregated columns
+        let alloc = Bump::new();
+        let evaluated_columns: HashMap<AggregationOperator, Vec<(Identifier, OwnedColumn<S>)>> =
+            self.aggregation_expr_map
+                .iter()
+                .map(
+                    |((agg_op, expr), id)| -> PostprocessingResult<(
+                        AggregationOperator,
+                        (Identifier, OwnedColumn<S>),
+                    )> {
+                        let evaluated_owned_column = owned_table.evaluate(expr)?;
+                        Ok((*agg_op, (*id, evaluated_owned_column)))
+                    },
+                )
+                .collect::<PostprocessingResult<Vec<_>>>()?
+                .into_iter()
+                .into_group_map();
```

Comment:
`process_results` would prevent `collect`ing an extra `Vec`.
```suggestion
                .process_results(|iter| iter.into_group_map())?;
```